### PR TITLE
Fix regressions from the JSON encoder/decoder rewrite

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: formats/json/commonTest/src/kotlinx/serialization/test/TestingFramework.kt [hunk 1]
```diff
@@ -38,10 +38,10 @@ inline fun <reified T : Any> StringFormat.assertStringFormAndRestored(
     assertEquals(original, restored)
 }
 
-inline fun <reified T : Any> assertSerializedAndRestored(
+fun <T : Any> assertSerializedAndRestored(
```
Comment (sandwwraith) [inline formats/json/commonTest/src/kotlinx/serialization/test/TestingFramework.kt:41]: Why this change is needed here?

Comment (qwwdfsad) [reply formats/json/commonTest/src/kotlinx/serialization/test/TestingFramework.kt:41]: well, `inline reified` is not necessary for this function at all, but prevent IDEA debugger from working (AFAIR breakpoints didn't hit properly)

Comment (sandwwraith) [reply formats/json/commonTest/src/kotlinx/serialization/test/TestingFramework.kt:41]: Ok then

**Review by sandwwraith — APPROVED** [review]

**Comment by sandwwraith** [comment]
Please preserve all commit messages with issue numbers
